### ISSUE TITLE
[C++] BPF::attach_uprobe() does not support specifying symbol name as regexp

### ISSUE THREAD

**Issue by yzhao1012** [iovisor/bcc#2553]
So it seems python API [1] allows to specify sym_re.

Can we backport this into C++ BPF API? If so, I can contribute.

Would like to learn any comments or guidance first.

[1] https://github.com/iovisor/bcc/blob/183be9a470035f1916349a29f31070400e9d3a70/src/python/bcc/__init__.py#L976

**Comment by palmtenor** [comment]
I honestly really don't like the regex symbol name. It makes a lot of things hidden in the resolution and makes debugging less obvious. If there is no compelling reason that some use case can not be done by users themselves I really prefer not adding the regex resolution to C++. Unless others disagree:)

**Comment by yzhao1012** [comment]
Thanks!
I think you prefer users to list the symbols and find the correct one. Follow up on that, would you be open to refactoring into BCC libraries, such that some of the internal functions can be easier reused for such purpose?

(I don't have a firm idea on what should be done in the refactorings, but just from my impressions, most of code is not well suited for direct customer use to list symbols and search, etc.)

**Comment by palmtenor** [comment]
Of course, what refactor do you have in mind that would help you?

**Comment by yzhao1012** [comment]
* We were originally thinking about adding the capability to do demangling inside BCC API.
* We have since decided to do mangling instead of that, to directly plug with the BCC APIs.
* The connection was that we originally were thinking to use regexp to match non-mangled name with symbols. We discovered that mangling is a well documented procedure that we can rely upon, so the regexp part is no longer in the requirements.

So for now, I think we no longer have incentive to pursue any refactoring. :)
